examples/webgpu_shadowmap_csm.html

```diff
@@ -0,0 +1,320 @@
+
+			const params = {
+				orthographic: false,
+				fade: false,
+				shadows: true,
+				maxFar: 1000,
+				mode: 'practical',
+				lightX: - 1,
+				lightY: - 1,
+				lightZ: - 1,
+				margin: 100,
+				shadowNear: 1,
+				shadowFar: 2000,
+				autoUpdateHelper: true,
+				updateHelper: function () {
+
+					csmHelper.update();
+
+				}
+			};
+
+			init();
+
+			function updateOrthoCamera() {
+
+				const size = controls.target.distanceTo( camera.position );
+				const aspect = camera.aspect;
+
+				orthoCamera.left = size * aspect / - 2;
+				orthoCamera.right = size * aspect / 2;
+
+				orthoCamera.top = size / 2;
+				orthoCamera.bottom = size / - 2;
+				orthoCamera.position.copy( camera.position );
+				orthoCamera.rotation.copy( camera.rotation );
+				orthoCamera.updateProjectionMatrix();
+
+			}
+
+			function init() {
+
+				scene = new THREE.Scene();
+				scene.background = new THREE.Color( '#454e61' );
+				camera = new THREE.PerspectiveCamera( 70, window.innerWidth / window.innerHeight, 0.1, 5000 );
+				orthoCamera = new THREE.OrthographicCamera();
+
+				renderer = new THREE.WebGPURenderer( { antialias: true } );
+				renderer.setSize( window.innerWidth, window.innerHeight );
+				renderer.setAnimationLoop( animate );
+				document.body.appendChild( renderer.domElement );
+				renderer.shadowMap.enabled = params.shadows;
+				renderer.shadowMap.type = THREE.PCFSoftShadowMap;
+
+				controls = new OrbitControls( camera, renderer.domElement );
+				controls.maxPolarAngle = Math.PI / 2;
+				camera.position.set( 60, 60, 0 );
+				controls.target = new THREE.Vector3( - 100, 10, 0 );
+				controls.update();
+
+				const ambientLight = new THREE.AmbientLight( 0xffffff, 1.5 );
+				scene.add( ambientLight );
+
+				const additionalDirectionalLight = new THREE.DirectionalLight( 0x000020, 1.5 );
+				additionalDirectionalLight.position.set( params.lightX, params.lightY, params.lightZ ).normalize().multiplyScalar( - 200 );
+				scene.add( additionalDirectionalLight );
+
+				csmDirectionalLight = new THREE.DirectionalLight( 0xffffff, 3.0 );
+
+				csmDirectionalLight.castShadow = true;
+				csmDirectionalLight.shadow.mapSize.width = 2048;
+				csmDirectionalLight.shadow.mapSize.height = 2048;
+				csmDirectionalLight.shadow.camera.near = params.shadowNear;
+				csmDirectionalLight.shadow.camera.far = params.shadowFar;
+				csmDirectionalLight.shadow.camera.top = 1000;
+				csmDirectionalLight.shadow.camera.bottom = - 1000;
+				csmDirectionalLight.shadow.camera.left = - 1000;
+				csmDirectionalLight.shadow.camera.right = 1000;
+				csmDirectionalLight.shadow.bias = - 0.001;
+
+				csm = new CSMShadowNode( csmDirectionalLight, { cascades: 4, maxFar: params.maxFar, mode: params.mode } );
+
+				csmDirectionalLight.position.set( params.lightX, params.lightY, params.lightZ ).normalize().multiplyScalar( - 200 );
+
+				csmDirectionalLight.shadow.shadowNode = csm;
```

Comment:
Is this line used? I'm trying to figure out where this `shadowNode` property is used in order to figure out if I'm missing something in the types for this.

Reply:
It is used:

https://github.com/mrdoob/three.js/blob/ed9f5a1977d356e4f6c01ccbe89521e310273db3/src/nodes/lighting/AnalyticLightNode.js#L60

As the name suggests it is type Node() but the typescript annotations probably need updating by whoever maintains them.

Reply:
Thanks! That would be me. :)